utils/prometheus_metrics.go

```diff
@@ -0,0 +1,48 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type ClientMetrics struct {
+	mobileMetrics *prometheus.HistogramVec
+}
+
+func NewClientMetrics() *ClientMetrics {
+	histogramMetrics := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "mobile_metrics_seconds",
+			Help:    "Mobile metrics",
+			Buckets: prometheus.ExponentialBuckets(0.01, 1.3, 36),
+		},
+		[]string{"event", "country", "app_version", "server_version"},
+	)
+	prometheus.MustRegister(histogramMetrics)
```

Comment:
what happens for failure cases where call was not successful

Reply:
MustRegister doesnt return any error. I am handling panic case using recover.